qiskit/assembler/assemble_schedules.py

```diff
@@ -92,9 +93,18 @@ def _assemble_experiments(
     instruction_converter = instruction_converter(qobj.PulseQobjInstruction,
                                                   **run_config.to_dict())
 
-    schedules = [
-        sched if isinstance(sched, pulse.Schedule) else pulse.Schedule(sched) for sched in schedules
-    ]
+    def schedule_converter(sched):
+        if isinstance(sched, pulse.Schedule):
+            return sched
+        elif isinstance(sched, ScheduleComponent):
+            new_sched = pulse.Schedule()
+            for t0, instr in sched.instructions:
+                new_sched.insert(t0, instr, inplace=True)
+            return new_sched
+        else:
+            raise QiskitError('Invalid parameter schedules: {}.'.format(type(sched)))
```

Comment:
Should we bother checking or can we assume (as it was doing before) that's either a  `Schedule` or a `ScheduleComponent`?